apps/files/src/fileactions.js

```diff
@@ -25,21 +27,38 @@ export default {
         {
           icon: 'file_download',
           handler: this.downloadFile,
-          ariaLabel: this.$gettext('Download'),
+          ariaLabel: (item) => {
+            return this.$gettext('Download')
+          },
           isEnabled: function (item) {
             return item.canDownload()
           }
         },
         {
           icon: 'delete',
-          ariaLabel: this.$gettext('Delete'),
           handler: this.deleteFile,
+          ariaLabel: (item) => {
+            return this.$gettext('Delete')
+          },
           isEnabled: function (item, parent) {
             if (parent && !parent.canBeDeleted()) {
               return false
             }
             return item.canBeDeleted()
           }
+        },
+        {
+          icon: 'star',
+          handler: this.toggleFileFavorite,
+          ariaLabel: (item) => {
+            if (item.starred) {
+              return this.$gettext('Unmark as favorite')
+            }
+            return this.$gettext('Mark as favorite')
+          },
+          isEnabled: function () {
+            return true
```

Comment:
Needs to be disabled when the user is not authorised.

Reply:
Done